### DISCUSSION TITLE
Why is the source code in one directory instead of properly organized just like modern mcp?

### DISCUSSION THREAD

**Discussion by brambora69123** [MCPHackers/RetroMCP-Java#149] [Q&A]
Just why? why is it not like modern mcp? its all in one directory and its harder to work with.

**Comment by PhoenixVX** [comment]
Because that is how MCP mappings are in these versions... not to mention how much of an issue it would be to reobfuscate back. Unless the packages matched the original structure of MC, access issues would become prevalent in every class that is inside that package.

Nothing else, you can always make your own mappings. There's nothing stopping you from mapping it yourself and setting up a filehost to host updated/custom mappings.

**Reply by Lassebq** [reply]
If only RMCP was originally designed to work with intermediary mappings in mind...
